### ISSUE TITLE
Does the multi-task training include predicting relevant paragraphs?

### ISSUE THREAD

**Issue by Fan-Luo** [allenai/longformer#156]
Hi,
The page mention in section 6.1
> We use a two-stage model that first selects the most relevant paragraphs then passes them to a second stage for answer extraction. Both stages concatenate question and context into one sequence, run it through Longformer, then use task-specific prediction layers. We train the models in a multi-task way to predict relevant paragraphs, evidence sentences, answer spans and question types (yes/no/span) jointly.

My understanding is that the first stage concatenates question and the whole context as input to Longformer, and only predict relevant paragraphs as a binary classification task for each paragraph. The second stage concatenates question and the predict relevant paragraphs as input, train another Longformer model in a multi-task way to predict **<s>relevant paragraphs,</s>** evidence sentences, answer spans and question types (yes/no/span) jointly.

Does the multi-task training in stage2 still need to include predicting relevant paragraphs?

Thank you

**Comment by armancohan** [comment]
Thanks for reaching out! 
Both models are identical (they both predict relevant paragraphs, evidence sentences, answer spans, and question types). 
The difference is that the second stage model uses up to 5 high scoring paragraphs from the first stage model (the paragraph score should be higher than a threshold as mentioned in the appendix).

**Comment by Fan-Luo** [comment]
Thank you for your clarification!

When you said 
> they both predict relevant paragraphs, evidence sentences, answer spans, and question types

I believe you meant they both compute the loss of all the tasks. Do the losses of the two stages added as `loss1 + loss2` before backprop?
Does decoding during inference only uses the prediction from stage2?

I also have a question about
>  For answer span extraction we use BERT’s QA model (Devlin et al., 2019) with addition of a question type (yes/no/span) classification head over the first special token ([CLS]).

Could you please explain `we use BERT’s QA model`? I thought both stages uses Longformer, then how you apply BERT after sequence_outputs from Longformer?
  
Thank you

**Comment by Fan-Luo** [comment]
Thank you!

One more question I would like to ask for clarification:

> For answer span extraction we use BERT’s QA model (Devlin et al., 2019) with addition of a question type (yes/no/span) classification head over the first special token ([CLS]).

I believe it means predicting answer span and question type in the similar way as did in BERT’s QA model (Devlin et al., 2019), instead of actually using BERT. 
Could you please correct me if my understanding is wrong?

Best
